homeassistant/components/homewizard_climate/climate.py

```diff
@@ -0,0 +1,226 @@
+
+    @property
+    def supported_features(self) -> ClimateEntityFeature:
+        """Return the list of supported features."""
+        return (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.FAN_MODE
+            | ClimateEntityFeature.SWING_MODE
+        )
+
+    @property
+    def swing_modes(self) -> list[str]:
+        """Todo."""
+        return [SWING_HORIZONTAL, SWING_OFF]
+
+    @property
+    def swing_mode(self) -> str:
+        """Todo."""
+        return (
+            SWING_HORIZONTAL
+            if self._device_web_socket.last_state.oscillate
+            else SWING_OFF
+        )
+
+    @property
+    def hvac_mode(self):
+        """Return hvac target hvac state."""
+        if self._device_web_socket.last_state.power_on:
+            result = (
+                HVACMode.HEAT
+                if self._device_web_socket.last_state.heater
+                else HVACMode.COOL
+            )
+        else:
+            result = HVACMode.OFF
+
+        return result
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available operation modes."""
+        return [HVACMode.HEAT, HVACMode.COOL, HVACMode.OFF]
+
+    @property
+    def temperature_unit(self):
+        """Todo."""
+        return TEMP_CELSIUS
+
+    @property
+    def target_temperature_step(self) -> float:
+        """Todo."""
+        return 1
+
+    @property
+    def target_temperature_high(self) -> float:
+        """Todo."""
+        return 30
+
+    @property
+    def target_temperature_low(self) -> float:
+        """Todo."""
+        return 14
+
+    @property
+    def min_temp(self) -> float:
+        """Todo."""
+        return self.target_temperature_low
+
+    @property
+    def max_temp(self) -> float:
+        """Todo."""
+        return self.target_temperature_high
+
+    @property
+    def target_temperature(self) -> float:
+        """Todo."""
+        return self._device_web_socket.last_state.target_temperature
+
+    def set_temperature(self, **kwargs) -> None:
+        """Todo."""
+        self._device_web_socket.set_target_temperature(
+            int(kwargs.get(ATTR_TEMPERATURE, "0"))
+        )
+
+    def set_humidity(self, humidity: int) -> None:
+        """Todo."""
+        raise NotImplementedError()
+
+    def set_fan_mode(self, fan_mode: str) -> None:
+        """Todo."""
+        if fan_mode == FAN_ON:
+            self._device_web_socket.turn_on()
+        elif fan_mode == FAN_OFF:
+            self._device_web_socket.turn_off()
+        elif fan_mode == FAN_LOW:
+            self._device_web_socket.set_fan_speed(1)
+        elif fan_mode == FAN_MEDIUM:
+            speed = 4 if self.hvac_mode == HVACMode.COOL else 2
+            self._device_web_socket.set_fan_speed(speed)
+        elif fan_mode == FAN_HIGH:
+            speed = 8 if self.hvac_mode == HVACMode.COOL else 4
+            self._device_web_socket.set_fan_speed(speed)
```

Comment:
It would be nice if state changes are batched/debounced to be a bit nicer to the server. Maybe add this to the API implementation itself.

Reply:
Good suggestion, I can bake a debouncing mechanism into the external library.
Do you see this as a blocker for this PR though?